### ISSUE TITLE
Add option to apply reduction operations also to coordinates instead of dropping them.

### ISSUE THREAD

**Issue by JulianGiles** [pydata/xarray#8317]
### Is your feature request related to a problem?

Applying reduction operations like .mean() to a Dataset or DataArray will remove all coordinates that have the reduced dimension.

### Describe the solution you'd like

I would like a keyword argument (reduce_coords?) in the reduction method to allow for the reduction to also be applied to the coordinates.

### Describe alternatives you've considered

A current solution is to promote the coordinates to variables, perform the reduction and then demote them back to coordinates, like in #1497 . But it would be nice to have a faster way of doing it, like in #3033 with keepdims=True to retain the dimensions after a reduction.

**Comment by TomNicholas** [comment]
This seems useful! Would it perhaps make sense to simply change the behaviour of `keepdims` to not drop coordinates which depend on the reduced dimension? That would be a breaking change though...

**Comment by JulianGiles** [comment]
Yes, that could also work. Although (in my case) I only need the reduced coordinate without the reduced dimension, but it is still much easier to just take out the reduced dimension after the operation than the workaround I mentioned in the post. However, I do not know what is easier to implement.
